source/calculate.cpp

```diff
@@ -61,21 +59,20 @@ namespace calculate {
             switch (current->type) {
                 case (Type::CONSTANT):
                 case (Type::RIGHT):
-                    if (next->is(Type::RIGHT) ||
-                        next->is(Type::SEPARATOR) ||
-                        next->is(Type::OPERATOR))
-                        break;
+                    if (next->is(Type::RIGHT)) break;
+                    else if (next->is(Type::SEPARATOR)) break;
+                    else if (next->is(Type::OPERATOR)) break;
+                    else throw SyntaxErrorException();
                 case (Type::LEFT):
                 case (Type::SEPARATOR):
                 case (Type::OPERATOR):
-                    if (next->is(Type::CONSTANT) ||
-                        next->is(Type::LEFT) ||
-                        next->is(Type::FUNCTION))
-                        break;
+                    if (next->is(Type::CONSTANT)) break;
+                    else if (next->is(Type::LEFT)) break;
+                    else if (next->is(Type::FUNCTION)) break;
+                    else throw SyntaxErrorException();
                 case (Type::FUNCTION):
-                    if (next->is(Type::LEFT))
-                        break;
-                    throw SyntaxErrorException();
```

Comment:
Line 78 was never reached what made the subroutine almost useless. This commit fixes #22.